src/Maker/MakeAuthenticator.php

```diff
@@ -55,50 +70,94 @@ public function configureCommand(Command $command, InputConfiguration $inputConf
+            'Which authentication type do you want ?',
+            array_keys($authenticatorTypeValues),
+            key($authenticatorTypeValues)
+        );
+        $input->setArgument(
+            'authenticator-type',
+            $authenticatorTypeValues[$authenticatorType]
+        );
 
-        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents($path));
+        $manipulator = new YamlSourceManipulator($this->fileManager->getFileContents('config/packages/security.yaml'));
         $securityData = $manipulator->getData();
 
-        $interactiveSecurityHelper = new InteractiveSecurityHelper();
-
-        $command->addOption('firewall-name', null, InputOption::VALUE_OPTIONAL, '');
-        $input->setOption('firewall-name', $firewallName = $interactiveSecurityHelper->guessFirewallName($io, $securityData));
-
-        $command->addOption('entry-point', null, InputOption::VALUE_OPTIONAL);
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')
+            && !isset($securityData['security']['providers']) || !$securityData['security']['providers']) {
+            throw new RuntimeCommandException('You need to have at least one provider defined in security.yaml');
+        }
 
-        $authenticatorClassNameDetails = $this->generator->createClassNameDetails(
-            $input->getArgument('authenticator-class'),
-            'Security\\'
+        // authenticator class
+        $command->addArgument('authenticator-class', InputArgument::REQUIRED);
+        $questionAuthenticatorClass = new Question('The class name of the authenticator to create (e.g. <fg=yellow>AppCustomAuthenticator</>)');
+        $questionAuthenticatorClass->setValidator(
+            function ($answer) {
+                Validator::notBlank($answer);
+
+                return Validator::classDoesNotExist(
+                    $this->generator->createClassNameDetails(
+                        $answer,
+                        'Security\\'
+                    )->getFullName()
+                );
+            }
         );
+        $input->setArgument('authenticator-class', $io->askQuestion($questionAuthenticatorClass));
 
+        $command->addOption('firewall-name', null, InputOption::VALUE_OPTIONAL);
+        $input->setOption('firewall-name', $firewallName = InteractiveSecurityHelper::guessFirewallName($io, $securityData));
+
+        $command->addOption('entry-point', null, InputOption::VALUE_OPTIONAL);
         $input->setOption(
             'entry-point',
-            $interactiveSecurityHelper->guessEntryPoint($io, $securityData, $authenticatorClassNameDetails->getFullName(), $firewallName)
+            InteractiveSecurityHelper::guessEntryPoint($io, $securityData, $input->getArgument('authenticator-class'), $firewallName)
         );
+
+        if (self::AUTH_TYPE_FORM_LOGIN === $input->getArgument('authenticator-type')) {
+            $command->addArgument('controller-class', InputArgument::OPTIONAL);
```

Comment:
Optional? I don't think it matters, because we're always passing it. But you use required above - let's be consistent. This looks required :).